src/JuMP.jl

```diff
@@ -933,24 +933,57 @@ end
 
 """
     get_optimizer_attribute(
-        model::Model, attr::MOI.AbstractOptimizerAttribute
+        model::Model,
+        attr::MOI.AnyAttribute,
+        args...
     )
 
 Return the value of the solver-specific attribute `attr` in `model`.
 
-## Example
+## Examples
 
+Query model or optimizer attributes:
 ```julia
 get_optimizer_attribute(model, MOI.Silent())
 ```
 
+Query variable or constraint attributes:
+```julia
+get_optimizer_attribute(model, MOI.VariablePrimalStart(), x)
+```
+
 See also: [`set_optimizer_attribute`](@ref), [`set_optimizer_attributes`](@ref).
 """
+function get_optimizer_attribute(model::Model, attr::MOI.AnyAttribute, args...)
+    # If we're in direct mode, no need for `AttributeFromOptimizer`.
+    if mode(model) == DIRECT
+        return MOI.get(model, attr, args...)
+    end
+    # ... else we have a caching optimizer.
+    state = MOI.Utilities.state(backend(model))
+    if state == MOI.Utilities.NO_OPTIMIZER
+        # There's no optimizer, so we can just query this from the model cache.
+        return MOI.get(model, attr, args...)
+    elseif state == MOI.Utilities.EMPTY_OPTIMIZER
+        # There's an optimizer, but it's not attached. Since the user might be
+        # querying some low-level attribute of a variable or constraint, we need
+        # to attach the optimizer.
+        MOI.Utilities.attach_optimizer(model)
```

Comment:
This won't work for solvers only implementing `optimize!`

Reply:
The alternative is to throw a nice error and encourage users to call `optimize!` or `MOI.Utilities.attach_optimizer`. Perhaps that's better.

Reply:
How about this now?